1-foundations/project-1-schelling-model/Project 1 - Schelling's model of segregation.ipynb

```diff
@@ -0,0 +1,676 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![moving](img/moving-schelling.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The simulation of the model is divided in rounds. When a round starts, the model iterates once through every agent evaluating their satisfaction and moving them to another spot if it applies. A round finishes when all of the agents have been looped through once, executing all the moves that applied. A new round starts when this process is repeated. The simulation finishes when the city achieves convergence (no one moved in an entire round) or a maximum number of rounds have been finalized.\n",
+    "\n",
+    "You might note that as a result of the moves in a round some agents that were initially dissatisfied will become satisfied as their neighborhoods change, or vice versa. We will not take care of these cases immediately, but in the next round of the model simulation. The instructions for that part of the implementation will be explained below."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Part 2: Python implementation\n",
+    "\n",
+    "This is how our specific implementation of Schelling's model will work:\n",
+    "\n",
+    "1. The initial state of the city (the grid) will be read from pre-defined files\n",
+    "1. When a round of the simulation begins, start the iteration with the square at the top left and perform the following operations:\n",
+    "    - Evaluate if the agent is dissatisfied\n",
+    "    - If they are:\n",
+    "        - Move the agent to its corresponding lot applying the moving rule\n",
+    "        - Update the city\n",
+    "1. Continue with the next square of the row and repeat the process\n",
+    "1. When the rows finishes, continue with the next row\n",
+    "1. When all the rows are looped over, perform the following checks:\n",
+    "    - If no one moved in the round, finish the simulation\n",
+    "    - If more than T rounds have passed, finish the simulation\n",
+    "1. If the simulation didn't finish, continue with a new round of the model\n",
+    "\n",
+    "And this is the moving rule we'll use:\n",
+    "\n",
+    "- An agent will move to another location that has a higher or equal similarity than the acceptable threshold. If no location has a higher or equal similarity, they will not move\n",
+    "- If more than one location is acceptable, they will move to the closest one in terms of Euclidean distance (formula is below)\n",
+    "- If more than one acceptable location is the closest, they will move to the one that has the highest similarity\n",
+    "- If still more than one is a candidate location, they will be indifferent between them and will choose any\n",
```

Comment:
What if the agent is dissatisfied but there are no locations where the acceptable threshold is met?